### PR TITLE
Bug 1782467: pull in openshift/library changes, including nginx/perl deprecations

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan 

@siamaksade FYI

this pulls in the recent deprecations

after this merges, we can cherry pick to 4.3

### CODE REVIEW

**Comment by gabemontero** [comment]
install flakes

/test e2e-aws-operator
/test e2e-aws-upgrade

**Comment by adambkaplan** [comment]
@gabemontero note that if you want these in 4.3 you'll need BZs for this PR and the backport

**Comment by gabemontero** [comment]
> @gabemontero note that if you want these in 4.3 you'll need BZs for this PR and the backport

yep understood .... per our email thread, I'm hoping @siamaksade will open the bugzilla to "officially request" getting this into 4.3 :-)

I'll ping him on slack if he doesn't respond in a day or two, and will retroactively open a bugzillas and associate here if needed.

**Comment by gabemontero** [comment]
install flakes

/retest

**Comment by gabemontero** [comment]
mco install flake

/retest

**Comment by gabemontero** [comment]
MCO pr has merged

/retest

**Comment by gabemontero** [comment]
/hold

blah ... we have extended tests trying to use the deprecated imagestreams:

`The ImageStreamTag "php:7.0" is invalid: from: Error resolving ImageStreamTag php:7.0 in namespace openshift: unable to find latest tagged image`

is from our builds tests, and jenkins is using redis and times out (they don't dump the deployment logs the where's there is smoke ...):
```
STEP: instantiate the test application
Dec 11 20:05:45.950: INFO: Running 'oc --namespace=e2e-test-jenkins-pipeline-t68sm --config=/tmp/configfile687957516 new-app redis-ephemeral --name redis -p MEMORY_LIMIT=128Mi'
--> Deploying template "openshift/redis-ephemeral" to project e2e-test-jenkins-pipeline-t68sm

     Redis (Ephemeral)
     ---------
     Redis in-memory data structure store, without persistent storage. For more information about using this template, including OpenShift considerations, see https://github.com/sclorg/redis-container/blob/master/5.
```

gonna have to update openshift/origin first

**Comment by gabemontero** [comment]
the jenkins redis failure is a bug in the SCL template/imagestream ... I'll be reverting that change here

**Comment by gabemontero** [comment]
the cakephp templates still ref php:7.1 ... reverting the recent php imagestream update

**Comment by gabemontero** [comment]
Peeling the onion apparently.  image-eco is not revealing that a perl test template references a deprecated imagestream:

```
--> Deploying template "e2e-test-s2i-perl-59k22/perl-hotdeploy-example" for "/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy/perl.json" to project e2e-test-s2i-perl-59k22

     * With parameters:
        * Name=perl

--> Creating resources ...
    service "perl" created
    imagestream.image.openshift.io "perl" created
    buildconfig.build.openshift.io "perl" created
    deploymentconfig.apps.openshift.io "perl" created
--> Success
    Use 'oc start-build perl' to start a build.
    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
     'oc expose svc/perl' 
    Run 'oc status' to view your app.
Dec 12 17:19:58.860: INFO: Running 'oc --namespace=e2e-test-s2i-perl-59k22 --config=/tmp/configfile086962793 start-build perl --from-dir=/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy -o=name'
Dec 12 17:20:00.243: INFO: Error running &{/usr/bin/oc [oc --namespace=e2e-test-s2i-perl-59k22 --config=/tmp/configfile086962793 start-build perl --from-dir=/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy -o=name] []    Uploading directory "/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy" as binary input for the build ...

Uploading finished
The ImageStreamTag "perl:5.24" is invalid: from: Error resolving ImageStreamTag perl:5.24 in namespace openshift: unable to find latest tagged image
 [] <nil> 0xc00522b1d0 exit status 1 <nil> <nil> true [0xc004d39218 0xc004d39240 0xc004d39250] [0xc004d39218 0xc004d39240 0xc004d39250] [0xc004d39220 0xc004d39238 0xc004d39248] [0x9d85f0 0x9d8720 0x9d8720] 0xc005292d20 <nil>}:
StdOut>

StdErr>
Uploading directory "/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy" as binary input for the build ...

Uploading finished
The ImageStreamTag "perl:5.24" is invalid: from: Error resolving ImageStreamTag perl:5.24 in namespace openshift: unable to find latest tagged image

Dec 12 17:20:00.243: INFO: 

start-build output with args [perl --from-dir=/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy -o=name]:
Error>exit status 1
StdOut>

StdErr>
Uploading directory "/tmp/fixture-testdata-dir181602482/test/extended/testdata/image_ecosystem/perl-hotdeploy" as binary input for the build ...

Uploading finished
The ImageStreamTag "perl:5.24" is invalid: from: Error resolving ImageStreamTag perl:5.24 in namespace openshift: unable to find latest tagged image
```

File: tmp/build/assets/operator/ocp-x86_64/nginx/imagestreams/nginx-rhel7.json [hunk 1]
```diff
@@ -11,47 +11,47 @@
         "tags": [
             {
                 "annotations": {
-                    "description": "Build and serve static content via Nginx HTTP server and a reverse proxy (nginx) on RHEL 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/nginx-container/blob/master/1.10/README.md.",
+                    "description": "Build and serve static content via Nginx HTTP server and a reverse proxy (nginx) on RHEL 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/nginx-container/blob/master/1.14/README.md.",
                     "iconClass": "icon-nginx",
                     "openshift.io/display-name": "Nginx HTTP server and a reverse proxy 1.10",
```
Comment (xiuwang) [inline tmp/build/assets/operator/ocp-x86_64/nginx/imagestreams/nginx-rhel7.json]: s/1.10/1.14

File: tmp/build/assets/operator/ocp-x86_64/nginx/imagestreams/nginx-rhel7.json [hunk 2]
```diff
@@ -11,47 +11,47 @@
...
                     "type": "Local"
                 }
             },
             {
                 "annotations": {
-                    "description": "Build and serve static content via Nginx HTTP server and a reverse proxy (nginx) on RHEL 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/nginx-container/blob/master/1.12/README.md.",
+                    "description": "Build and serve static content via Nginx HTTP server and a reverse proxy (nginx) on RHEL 7. For more information about using this builder image, including OpenShift considerations, see https://github.com/sclorg/nginx-container/blob/master/1.10/README.md.",
                     "iconClass": "icon-nginx",
-                    "openshift.io/display-name": "Nginx HTTP server and a reverse proxy 1.12",
+                    "openshift.io/display-name": "Nginx HTTP server and a reverse proxy 1.14",
```
Comment (xiuwang) [inline tmp/build/assets/operator/ocp-x86_64/nginx/imagestreams/nginx-rhel7.json]: s/1.14/1.10

**Pull request merged**